mkinitcpio

```diff
@@ -443,9 +453,25 @@ process_preset() (
             done
         fi
 
-        preset_cmd+=($OPTREST)
-        msg2 "${preset_cmd[*]}"
-        MKINITCPIO_PROCESS_PRESET=1 "$0" "${preset_cmd[@]}"
+        if (( _optremove )); then
+            if (( ${#preset_remove_cmd[*]} )); then
+                # we won't be calling mkinitcpio recursively, so no need to set MKINITCPIO_PROCESS_PRESET
+                for pr in "${preset_remove_cmd[@]}"; do
+                    if [[ ! -f "${pr}" ]]; then
+                        warning "Image not found: \`%s'" "$pr"
+                    elif [[ ! -w "${pr}" ]]; then
+                        error "Unable to delete image: \`%s'" "$pr"
```

Comment:
It may be better to explicitly say that it's not writable. "Unable to delete image" just makes one wonder "why?".

Reply:
Updated to:
```
==> ERROR: Image not writable: `/tmp/arch-linux.efi'
```